lib/MultiReasonException.php

```diff
@@ -11,7 +11,7 @@ class MultiReasonException extends \Exception {
      * @param string|null $message
      */
     public function __construct(array $reasons, string $message = null) {
-        parent::__construct($message ?: "Too many promises were rejected");
+        parent::__construct($message ?: "Multiple errors encountered");
```

Comment:
I think we should stringify at least the exception names and messages into the message, otherwise they get lost.